paper/paper.md

```diff
@@ -0,0 +1,48 @@
+---
+title: 'Setfield.jl: Changing the immutable'
+tags:
+  - Julia
+  - Functional Programming
+  - Optics
+authors:
+  - name: Jan Weidner
+    orcid: 0000-0002-0980-8239
+    affiliation: 1
+affiliations:
+ - name: PTW-Dosimetry
+   index: 1
+date: 13 September 2020
+bibliography: paper.bib
+---
+
+# Summary
+We discuss the problem of updating immutable objects. The solutions presented are implemented in the Setfield.jl package [@SetfieldPackage].
+
+# Overview
+
+In the Julia programming language [@Julia-2017], some objects are *mutable* (`Array`, `mutable struct`, `...`), while others are *immutable* (`Tuple`, `struct`, `...`).
+Neither is strictly better than the other in every situation. However, *immutability* usually leads to code that is easier to reason about, for both humans and compilers.
+And therefore less buggy and more performant programs.
+One convenience with mutability is, that it makes updating objects very simple:
+
+`spaceship.captain.name = "Julia"`
+
+The analogous operation in the immutable case is to create a copy of `spaceship`,
+with just the captains name changed to "Julia". This operation is sometimtes called functional update.
+Just think for a moment, how would you do achieve this?
+It is a non trivial problem and there are many approaches. Both in Julia [@JuliaPR21912; @MutabilitiesPackage] and other languages [@HaskellLens; @ImmutableJS].
```

Comment:
Mutabilities.jl is just a wrapper of non-type-changing lens from Setfield.jl. So I don't feel it's adequate for showing as an alternative "approach." It may be useful for briefly discussing it as an alternative interface for lenses, though.

Reply:
That is an interestig perspective. I thought of `Mutabilities` as 
```julia
melt(obj) = mutablecopy(obj)
freeze(obj) = immutablecopy(obj)
```
And to execute an update you would do `melt ... update! ... freeze`.
I thought of the partial `melts/freezes` as less important and passing between mutable and immutable as the key idea.
You seem to put much more emphasis on the partial aspect (which I agree is pretty close to lenses) and less emphasis on passing between mutable and immutable.

So really passing between mutable and immutable was the alternative approach I wanted to highlight.